src/color_chooser.rs

```diff
@@ -55,9 +52,13 @@ impl<O: IsA<ColorChooser>> ColorChooserExt for O {
         }
     }
 
-    fn add_palette(&self, orientation: Orientation, colors_per_line: i32, colors: Vec<GdkRGBA>) {
+    fn add_palette(&self, orientation: Orientation, colors_per_line: i32, colors: &mut [RGBA]) {
```

Comment:
Whoops, looks like that should have been an `as_ptr()` call, in which case you're correct.